app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -114,8 +115,75 @@ def assigned
       render json: { topics: serialize_data(topics, AssignedTopicSerializer) }
     end
 
+    def group_assignments
+      if SiteSetting.assign_enabled?
+        offset = (params[:page].to_i * 30 || 0).to_i
+        is_group = (params[:is_group] || 1).to_i
+
+        if is_group == 1
+          topics = Topic
+            .includes(:tags)
+            .includes(:user)
+            .joins("JOIN topic_custom_fields tcf ON topics.id = tcf.topic_id AND tcf.name = 'assigned_to_id' AND tcf.value IS NOT NULL")
+            .where("tcf.value IN (SELECT group_users.user_id::varchar(255) FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", params[:group_name])
+            .order('tcf.value::integer, topics.bumped_at desc')
+
+          load_more = topics.to_a.length > 30 * (params[:page].to_i + 1)
+
+          topics = topics
+            .offset(offset)
+            .limit(30)
+
+          users = User.where("users.id IN (SELECT group_users.user_id FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", params[:group_name])
+        else
+          users = User.where("username = ?", params[:group_name])
```

Comment:
Why are we using the `group_name` parameter to describe a user? That is kind of confusing.

When requesting the topics for a single user, do we need to use this endpoint? Or could we re-use the existing per-user endpoint?

Reply:
We need to use this endpoint as existing one doesn't allow getting data of other users unless current_user is admin

Reply:
Oh I think that is probably a mistake. The assign plugin was originally for staff-only, but now we made it so you can customize the groups. So instead of checking admin, can you change that route to check for `current_user.can_assign`?